### PR TITLE
Add compile_args to Compiler._try_compile_tmp

### PR DESCRIPTION
**Pull request by slefrancois**
Compiler._try_compile_tmp did not use compile_args. The -m32 flag was missing in the gcc args for 32 bit platforms, preventing the correct BLAS from being linked.

### CODE REVIEW

**Comment by slefrancois** [comment]
jenkins test this

**Comment by nouiz** [comment]
There is many errors like:

Traceback (most recent call last):
  File "/usr/local/miniconda2/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/usr/local/miniconda2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-gpu0/workspace/Theano gpu/theano/sandbox/cuda/tests/test_abstractconv.py", line 36, in tcase
    if not dnn_available():
  File "/home/jenkins-gpu0/workspace/Theano gpu/theano/sandbox/cuda/__init__.py", line 321, in dnn_available
    try_run=False, output=True)
  File "/home/jenkins-gpu0/workspace/Theano gpu/theano/sandbox/cuda/nvcc_compiler.py", line 79, in try_flags
    nvcc_path)
  File "/home/jenkins-gpu0/workspace/Theano gpu/theano/gof/cmodule.py", line 1808, in _try_flags
    output=output, compiler=compiler)
  File "/home/jenkins-gpu0/workspace/Theano gpu/theano/gof/cmodule.py", line 1735, in _try_compile_tmp
    args = cls.compile_args()
AttributeError: type object 'Compiler' has no attribute 'compile_args'

File: .jenkins/jenkins_buildbot_python2_32bit.sh [hunk 1]
```diff
@@ -15,4 +15,4 @@ echo
 
 FILE=${BUILDBOT_DIR}/theano_python32bit_tests.xml
 set -x
-PYTHONPATH=$HOME/anaconda2_32bit/lib/python2.7/site-packages THEANO_FLAGS=device=cpu,force_device=true,lib.amdlibm=False,compiledir=$WORKSPACE/compile/theano_compile_dir_theano_python2_32bit,blas.ldflags= $HOME/anaconda2_32bit/bin/python bin/theano-nose ${THEANO_PARAM} ${XUNIT}${FILE}
+PYTHONPATH=$HOME/anaconda2_32bit/lib/python2.7/site-packages THEANO_FLAGS=device=cpu,force_device=true,lib.amdlibm=False,compiledir=$WORKSPACE/compile/theano_compile_dir_theano_python2_32bit $HOME/anaconda2_32bit/bin/python bin/theano-nose ${THEANO_PARAM} ${XUNIT}${FILE}
```
Comment (nouiz) [inline .jenkins/jenkins_buildbot_python2_32bit.sh]: I don't recall making this commit. Do you know why it is tagged as being done by me?

File: theano/gof/cmodule.py [hunk 2]
```diff
@@ -1711,8 +1711,8 @@ class Compiler(object):
 
     """
 
-    @staticmethod
-    def _try_compile_tmp(src_code, tmp_prefix='', flags=(),
+    @classmethod
+    def _try_compile_tmp(cls, src_code, tmp_prefix='', flags=(),
                          try_run=False, output=False, compiler=None):
         """
         Try to compile (and run) a test program.
```
Comment (nouiz) [inline theano/gof/cmodule.py]: Add in all changed method the docstring to tell that we prepend to the flags the compile_args flags

**Comment by nouiz** [comment]
The error is because you didn't changed the class NVCC_Compiler in the file cuda/nvcc_compiler.py

File: .jenkins/jenkins_buildbot_python2_32bit.sh (hunk 1 above)
Comment (slefrancois) [inline .jenkins/jenkins_buildbot_python2_32bit.sh]: I pushed it from tikuanyin, where git is setup with your username.

**Pull request merged**